exercises/triangle/examples/src/Triangle.purs

```diff
@@ -0,0 +1,41 @@
+module Triangle
+  ( Triangle(Equilateral, Isosceles, Scalene)
+  , triangleKind
+  ) where
+
+import Prelude
+import Data.Set as Set
+import Data.Either (Either(Left, Right))
+import Data.Generic (class Generic, gShow, gEq)
+
+
+data Triangle
+  = Equilateral
+  | Isosceles
+  | Scalene
+
+derive instance genericTriangle :: Generic Triangle
```

Comment:
I don't understand this statement and I couldn't find any info on it in the Purescript book. Could you point me in the direction of an explanation of `derive instance`? :)